runtime/runtime/src/actions.rs

```diff
@@ -233,8 +235,12 @@ pub(crate) fn action_deploy_contract(
     account_id: &AccountId,
     deploy_contract: &DeployContractAction,
 ) {
+    let account = account.as_mut().unwrap();
     let code = ContractCode::new(deploy_contract.code.clone());
-    account.as_mut().unwrap().code_hash = code.get_hash();
+    let prev_code = get_code(state_update, account_id);
+    account.storage_usage -= prev_code.map(|code| code.code.len() as u64).unwrap_or_default();
```

Comment:
should code also should pay for the key?

Reply:
key length is account name length, we could add extra fixed cost to punish long names but shouldn't matter too much because names have a limit.